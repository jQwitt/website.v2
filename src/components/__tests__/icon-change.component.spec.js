import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import IconChange from '../icon-change.component';

afterEach(cleanup);
describe('IconChange renders correctly', () => {
  test('snapshot', () => {
    const { asFragment } = render(<IconChange />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('conditionally renders className', () => {
    const data = {
      className: 'testClass',
    };
    const { container } = render(<IconChange className={data.className} />);
    expect(container.firstChild.classList.contains(data.className)).toBe(true);
  });
  it('toggles icon on click', () => {
    const data = {
      firstIcon: 'add',
      secondIcon: 'subTract',
    };
    const { container, getByTestId } = render(
      <IconChange firstIcon={data.firstIcon} secondIcon={data.secondIcon} />,
    );
    fireEvent.click(container.firstChild);
    expect(
      getByTestId('togglesIcon').classList.contains(
        `flaticon-${data.secondIcon}`,
      ),
    ).toBe(true);
    fireEvent.click(container.firstChild);
    expect(
      getByTestId('togglesIcon').classList.contains(
        `flaticon-${data.firstIcon}`,
      ),
    ).toBe(true);
  });
});
