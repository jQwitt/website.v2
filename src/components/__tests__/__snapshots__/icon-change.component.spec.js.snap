// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconChange renders correctly snapshot 1`] = `
<DocumentFragment>
  <div
    class=" icon"
  >
    <i
      class="flaticon-add medium"
      data-testid="togglesIcon"
    />
  </div>
</DocumentFragment>
`;
